src/lxml/includes/etree_defs.h

```diff
@@ -267,6 +267,12 @@ static void* lxml_unpack_xmldoc_capsule(PyObject* capsule, int* is_owned) {
     xmlDoc *c_doc;
     void *context;
     *is_owned = 0;
+    if (unlikely_condition(!PyCapsule_IsValid(capsule, (const char*)"libxml2:xmlDoc"))) {
```

Comment:
I prefer to be explicit since the documentation of GetPointer does not state that it checks the type of its argument, I dont want to rely on it.